src/QirTools/qirlib/src/emit/qir/basic_values.rs

```diff
@@ -18,6 +21,17 @@ pub(crate) fn u64_to_i32<'ctx>(context: &Context<'ctx>, value: u64) -> BasicValu
         .as_basic_value_enum()
 }
 
+pub(crate) fn i64_to_i32<'ctx>(context: &Context<'ctx>, value: i64) -> BasicValueEnum<'ctx> {
```

Comment:
Odd thought, but would a new `trait IntoBasicValue<'ctx, T> { fn into_basic_value(context: &Context<'ctx>) -> BasicValueEnum<'ctx>; }` be helpful? Could then group these functions into something like `impl<'ctx> IntoBasicValue<'ctx, i32> for i64` blocks?